app/controllers/application_controller/filter.rb

```diff
@@ -745,10 +743,22 @@ def build_listnav_search_list(db)
 
     temp = MiqSearch.new
     temp.description = _("ALL")
+    temp.name = _("ALL")
     temp.id = 0
     @def_searches = MiqSearch.where(:db => [db, db.constantize.to_s]).visible_to_all.sort_by { |s| s.description.downcase }
     @def_searches = @def_searches.unshift(temp) unless @def_searches.empty?
+
+    @def_searches.each do |search|
+      search.description = search_description(search)
+      @selected_filter = search if def_searches_active_filter?(search) == 'active'
+    end
     @my_searches = MiqSearch.where(:search_type => "user", :search_key => session[:userid], :db => [db, db.constantize.to_s]).sort_by { |s| s.description.downcase }
+    if !@my_searches.empty?
```

Comment:
This is unnecessary, because `@my_searches.each` with an empty Array with just not execute the block, which is effectively the same thing.